src/core/expr/fexpr_nth.cc

```diff
@@ -44,7 +44,7 @@ class FExpr_Nth : public FExpr_Func {
         for (size_t i = 0; i < wf.ncols(); ++i) {
           Column coli = wf.retrieve_column(i);
           coli = evaluate1(std::move(coli), n_, gby);
-          outputs.add_column(std::move(coli), std::string(), Grouping::GtoONE);
+          outputs.add_column(std::move(coli), std::string(wf.retrieve_name(i)), Grouping::GtoONE);
```

Comment:
@oleksiyskononenko the grouping is GtoONE, unlike the others that are GtoALL, so I expect a size mismatch error ?

Reply:
>  // GtoONE
  //   Each group is mapped to exactly 1 row. This grouping mode is
  //   common as a result of "reduce" operations such as `sum`, `sd`,
  //   `min`, etc. A column with this level may need to be expanded
  //   in order to become conformable with any full-sized column.

So it's fine to have this grouping mode and it may be expanded if needed.

Reply:
ok, I'm not sure how to implement that. do you mind showing me how?

Reply:
Just implement it in exactly the same way as you did for `cumsum`, `cummin`, and all the other cumulative functions.

Reply:
I've been looking at this and let's keep it in the way it works now.